pyxem/data/_data.py

```diff
@@ -0,0 +1,311 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2023 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+import pooch
+from pyxem.data._registry import _file_names_hash, _urls
+import hyperspy.api as hs
+from pathlib import Path
+from pyxem.release_info import version
+from typing import Optional, Union
+import os
+
+# Create a downloader object
+marshall = pooch.create(
+    path=pooch.os_cache("pyxem"),
+    base_url="",
+    version=version.replace(".dev", "+"),
+    version_dev="develop",
+    env="PYXEM_DATA_DIR",
+    registry=_file_names_hash,
+    urls=_urls,
+)
+
+
+def au_grating(allow_download=False, **kwargs):
```

Comment:
We've chosen to describe the license of each dataset in the `kikuchipy.data` module. Might be something to consider for pyxem as well.

Reply:
Also, if datasets are hosted elsewhere than in the package, a link to the host can be added to docstring notes.

Reply:
To be honest I don't really know what the best way to handle the license for each dataset.  Some of them are mine but some are from different people contributing to the pyxes-demos.

Reply:
Done! (Well I should probably include the license in zenodo as well but this should be good here